lib/routes/account.js

```diff
@@ -549,6 +549,61 @@ module.exports = function (
     },
     {
       method: 'GET',
+      path: '/account/profile',
+      config: {
+        auth: {
+          mode: 'optional',
+          strategies: [
+            'sessionToken',
+            'oauthToken'
+          ]
+        }
+      },
+      handler: function (request, reply) {
+        var auth = request.auth
+        var uid
+        if (auth.strategy === 'sessionToken') {
+          uid = auth.credentials.uid
+        } else {
+          uid = Buffer(auth.credentials.user, 'hex')
+        }
+        function hasProfileItemScope(item) {
+          if (auth.strategy === 'sessionToken') {
+            return true
+          }
+          var scopes = auth.credentials.scopes
+          for (var i = 0; i < scopes.length; i++) {
+            if (scopes[i] === 'profile') {
```

Comment:
We discussed this a little IRL and it sounds like the right answer is to provide an abstraction for scope-checking in a shared utility lib of some kind.  But we don't have to do that in this PR.